python/pyfury/tests/test_meta_string.py

```diff
@@ -0,0 +1,199 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import pytest
+import numpy as np
+from pyfury.meta.metastring import (
+    MetaStringEncoder,
+    MetaStringDecoder,
+    Encoding,
+)
+
+
+def test_encode_meta_string_lower_special():
+    encoder = MetaStringEncoder("_", "$")
+    decoder = MetaStringDecoder("_", "$")
+    encoded = encoder._encode_lower_special("abc_def")
+    assert len(encoded) == 5
+    assert len(encoder.encode("org.apache.fury.benchmark.data").encoded_data) == 19
+    assert len(encoder.encode("MediaContent").encoded_data) == 9
+    decoded = decoder.decode(encoded, Encoding.LOWER_SPECIAL)
+    assert decoded == "abc_def"
+
+    for i in range(128):
+        builder = "".join(chr(ord("a") + j % 26) for j in range(i))
+        encoded = encoder._encode_lower_special(builder)
+        decoded = decoder.decode(encoded, Encoding.LOWER_SPECIAL)
+        assert decoded == builder
+
+
+def test_encode_meta_string_lower_upper_digit_special():
+    special_char1 = "."
+    special_char2 = "_"
+    encoder = MetaStringEncoder(special_char1, special_char2)
+    encoded = encoder._encode_lower_upper_digit_special("ExampleInput123")
+    assert len(encoded) == 12
+    decoder = MetaStringDecoder(special_char1, special_char2)
+    decoded = decoder.decode(encoded, Encoding.LOWER_UPPER_DIGIT_SPECIAL)
+    assert decoded == "ExampleInput123"
+
+    for i in range(1, 128):
+        string = create_string(i, special_char1, special_char2)
+        encoded = encoder._encode_lower_upper_digit_special(string)
+        decoded = decoder.decode(encoded, Encoding.LOWER_UPPER_DIGIT_SPECIAL)
+        assert decoded == string, f"Failed at {i}"
+
+
+def create_string(length, special_char1, special_char2):
+    chars = []
+    for j in range(length):
+        n = j % 64
+        if n < 26:
+            chars.append(chr(ord("a") + n))
+        elif n < 52:
+            chars.append(chr(ord("A") + (n - 26)))
+        elif n < 62:
+            chars.append(chr(ord("0") + (n - 52)))
+        elif n == 62:
+            chars.append(special_char1)
+        else:
+            chars.append(special_char2)
+    return "".join(chars)
+
+
+def test_meta_string():
```

Comment:
Could we rename this as `test_metastring`? And rename `test_meta_string.py` as `test_metastring.py`